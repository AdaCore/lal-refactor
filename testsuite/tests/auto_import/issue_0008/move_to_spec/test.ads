with Ada.Text_IO;

package Test is
   procedure Bar;
   procedure Foo (File : Ada.Text_IO.File_Type; U : Ada.Strings.Unbounded.Unbounded_String) renames Put_Line;
end Test;
