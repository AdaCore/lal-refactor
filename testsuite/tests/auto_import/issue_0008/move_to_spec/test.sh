auto_import -P default.gpr --source test.ads --line 5 --column 101
