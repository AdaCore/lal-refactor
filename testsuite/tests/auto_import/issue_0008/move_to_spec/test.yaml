description:
    This test checks that if the with clause is needed in the spec but it already exists
    in the body, it is moved from the body to the spec instead of being added.

driver: shell_script
