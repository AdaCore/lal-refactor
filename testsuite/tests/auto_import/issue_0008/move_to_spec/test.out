Available Imports:
Import: Ada.Strings.Unbounded.Text_IO; Qualifier: Ada.Strings.Unbounded.Text_IO
Import: Ada.Strings.Wide_Unbounded.Wide_Text_IO; Qualifier: Ada.Strings.Wide_Unbounded.Wide_Text_IO
Import: Ada.Strings.Wide_Wide_Unbounded.Wide_Wide_Text_IO; Qualifier: Ada.Strings.Wide_Wide_Unbounded.Wide_Wide_Text_IO
Import: Ada.Text_IO; Qualifier: Ada.Text_IO
Import: Ada.Text_IO.Unbounded_IO; Qualifier: Ada.Text_IO.Unbounded_IO
Import: Ada.Wide_Text_IO; Qualifier: Ada.Wide_Text_IO
Import: Ada.Wide_Text_IO.Wide_Unbounded_IO; Qualifier: Ada.Wide_Text_IO.Wide_Unbounded_IO
Import: Ada.Wide_Wide_Text_IO; Qualifier: Ada.Wide_Wide_Text_IO
Import: Ada.Wide_Wide_Text_IO.Wide_Wide_Unbounded_IO; Qualifier: Ada.Wide_Wide_Text_IO.Wide_Wide_Unbounded_IO
Import: GNAT.IO; Qualifier: GNAT.IO
Import: System.IO; Qualifier: System.IO
Import: Text_IO; Qualifier: Text_IO

Edits:
test.adb
1:1-2:1 
test.ads
1:1-1:1 with Ada.Strings.Unbounded.Text_IO;

5:101-5:101 Ada.Strings.Unbounded.Text_IO.

